### ISSUE TITLE
Treemap chart

### ISSUE THREAD

**Issue by chocstarfish** [novus/nvd3#1049]
Hi, I am looking for a treemap chart implementation, something like http://bl.ocks.org/mbostock/4063582 and http://mbostock.github.io/d3/talk/20111018/treemap.html.

Is it possible that this feature would be implemented in the near future? If not, can you give me a hint where I can get started in order to contribute this feature to nvd3?

Thanks.

**Comment by liquidpele** [comment]
It's not on the roadmap, but if you implement it feel free to send us a pull request.  

Basically if you clone our git repo, you can see each chart is a particular model in the src/models/ directory, you can create your own model and use the code from the blocks to make a new chart... I actually just did that with the sunburst chart :)

Once you make a model, you'll probably want to make an example in the examples directory and link to it in site.html with the others, so you can test it out.   All the stuff in examples links to the built files, which you build via:

https://github.com/novus/nvd3#building-latest

Don't really have time to walk you through much more than that... if you're proficient in javascript you should be able to pick it up though.  Cheers :)

**Comment by chocstarfish** [comment]
Thanks, I'll see what I can do.

**Comment by kfred** [comment]
+1 on treemap. has anyone built a model yet?

**Comment by liquidpele** [comment]
Nope, feel free to do it and send us a pull request ;)

**Comment by kfred** [comment]
Alrighty, I may actually contribute here. For once in my life I should stop being a leech ;)

I'm probably going to hack together something quick for a project I am working on, with the goal to eventually clean it and contribute a model for it.

Treechart would be a very nice addition to this library methinks.

**Comment by lucasxteixeira** [comment]
+1 on this!

**Comment by finbarmaginn** [comment]
+1

**Comment by kfred** [comment]
@finbarmaginn I finally got around to this a week ago. 

I have finished writing a "custom model" which creates reusable d3 treemap charts by passing in an array of key value pairs (and other various arguments unique to my project). I built this for a private project I am working on, which requires limited customization functionality, so my model largely hardcodes settings.

This month I will start looking at nvd3 model templates, per liquidpele's suggestion, and replicate one using my code. I now understand native d3 better and don't anticipate much additional work on my end, so I expect to submit a pull request for this nvd3 model this month.
